service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerTest.java

```diff
@@ -216,4 +204,85 @@ private CallbackRequest buildCallbackRequest() {
             .put("localAuthorityOrCafcass", LOCAL_AUTHORITY_NAME)
             .build();
     }
+
+    @Nested
+    class MidEvent {
+        private Document document;
+        private final byte[] pdf = {1, 2, 3, 4, 5};
+
+        @BeforeEach
+        void setUp() {
+            document = document();
+            DocmosisDocument docmosisDocument = new DocmosisDocument("order.pdf", pdf);
+
+            given(docmosisDocumentGeneratorService.generateDocmosisDocument(any(), any())).willReturn(docmosisDocument);
+            given(uploadDocumentService.uploadPDF(any(), any(), any(), any())).willReturn(document);
+        }
+
+        @Test
+        void midEventShouldGenerateOrderDocumentWhenOrderTypeIsBlankOrder() throws Exception {
+            AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(
+                generateCallbackRequest(BLANK_ORDER, false));
+
+            CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+            verify(uploadDocumentService, times(1)).uploadPDF(userId, userAuthToken, pdf, "blank_order_c21.pdf");
+
+            assertThat(caseData.getOrderTypeAndDocument().getDocument()).isEqualTo(expectedDocument());
+        }
+
+        @Test
+        void midEventShouldGenerateOrderDocumentWhenOrderTypeIsCareOrderWithFurtherDirections() throws Exception {
+            final AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(
+                generateCallbackRequest(CARE_ORDER, true));
+
+            verify(docmosisDocumentGeneratorService, times(1)).generateDocmosisDocument(any(), any());
+            verify(uploadDocumentService, times(1)).uploadPDF(userId, userAuthToken, pdf, "care_order.pdf");
+
+            final CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+            assertThat(caseData.getOrderTypeAndDocument().getDocument()).isEqualTo(expectedDocument());
+        }
+
+        @Test
+        void midEventShouldNotGenerateOrderDocumentWhenOrderTypeIsCareOrderWithNoFurtherDirections() throws Exception {
+            postMidEvent(generateCallbackRequest(CARE_ORDER, false));
+
+            verify(docmosisDocumentGeneratorService, never()).generateDocmosisDocument(any(), any());
+            verify(uploadDocumentService, never()).uploadPDF(any(), any(), any(), any());
+        }
+
+        @AfterEach
+        void tearDown() {
+            reset(docmosisDocumentGeneratorService);
+            reset(uploadDocumentService);
+        }
+
+        private CallbackRequest generateCallbackRequest(GeneratedOrderType orderType, boolean withFurtherDirections)
```

Comment:
I would almost prefer two separate methods with more specific naming as to what i'm getting returned rather than the use of a flag to augment data. But happy as it is.

Reply:
Can will change